src/chart-addons/stacked-legend.js

```diff
@@ -119,77 +156,93 @@ export function getLegendStateFromChart(chart, useMap, selectedLayer) {
+            scale => scale.name === color_scale_name
+          )
+          if (!materialized_color_scale) {
+            return {}
+          }
+
+          return {
+            ...materialized_color_scale,
+            legend: color_legend,
+            legend_position,
+            version: 2.0
+          }
+        } else {
+          // TODO(croot): this can be removed once all raster layer types are
+          // transitioned to the getPrimaryColorScaleName/getLegendDefinitionForProperty
+          // form above
+          const layerState = layer.getState()
+          const color = layer.getState().encoding.color
+
+          let color_legend_descriptor = null
 
-        return {
-          ...materialized_color_scale,
-          legend: color_legend,
-          legend_position,
-          version: 2.0
-        }
-      } else {
-        // TODO(croot): this can be removed once all raster layer types are
-        // transitioned to the getPrimaryColorScaleName/getLegendDefinitionForProperty
-        // form above
-        const layerState = layer.getState()
-        const color = layer.getState().encoding.color
-
-        let color_legend_descriptor = null
-
-        if (
-          (layers.length > 1 || _.isEqual(selectedLayer, layerState)) &&
-          typeof color !== "undefined"
-        ) {
           if (
-            typeof color.scale === "object" &&
-            color.scale.domain === "auto"
+            (layers.length > 1 || _.isEqual(selectedLayer, layerState)) &&
+            typeof color !== "undefined"
           ) {
-            color_legend_descriptor = {
-              ...color,
-              scale: {
-                ...color.scale,
+            if (
+              typeof color.scale === "object" &&
+              color.scale.domain === "auto"
+            ) {
+              color_legend_descriptor = {
+                ...color,
+                scale: {
+                  ...color.scale,
+                  domain: layer.colorDomain()
+                }
+              }
+            } else if (
+              typeof color.scale === "undefined" &&
+              color.domain === "auto"
+            ) {
+              color_legend_descriptor = {
+                ...color,
                 domain: layer.colorDomain()
               }
-            }
-          } else if (
-            typeof color.scale === "undefined" &&
-            color.domain === "auto"
-          ) {
-            color_legend_descriptor = {
-              ...color,
-              domain: layer.colorDomain()
+            } else if (color.type === "ordinal") {
+              const colValues = await getTopValues(layer, color.domain.length)
+              const { newDomain, newRange } = colValues
+                ? getUpdatedDomainRange(
+                    colValues,
+                    color.domain,
+                    color.range,
+                    color.defaultOtherRange
+                  )
+                : undefined
```

Comment:
This will try to destructure `undefined` and bomb if it hits the `else` condition
```suggestion
                : {}
```